.circleci/config.yml

```diff
@@ -0,0 +1,100 @@
+version: 2.1
+workflows:
+  version: 2
+  main:
+    jobs:
+      - check_format
+      - build_test
+
+jobs:
+  check_format:
+    machine:
+      image: ubuntu-1604:201903-01
+    environment:
+      - RUST_TEST_THREADS: "8"
+    steps:
+      - checkout
+      - restore_rustup_cache
+      - install_rust
+      - run:
+          name:  "Install format crates Rust"
+          command: |
+            rustup component add rustfmt
+            cargo install cargo-tomlfmt
+      - run:
+          name: "Run Git hooks"
+          command: |
+            source $HOME/.cargo/env
+            sh ./.githooks/pre-commit
```

Comment:
:question: Why do you need to run pre-commit hooks at a stage where nothing is about to be committed?

Reply:
If think we just wants to run the checks inside the commit hook :upside_down_face:

Reply:
It is now changed ;)